src/test/java/com/philips/research/bombar/core/domain/ProjectInteractorTest.java

```diff
@@ -210,6 +210,30 @@ void readsLicenseDistributionForProject() {
             assertThat(distribution).containsEntry("B", 1);
         }
 
+        @Test
+        void noObligationsForEmptyProject() {
+            project.clearDependencies();
+            final var ObligationsMap = interactor.getObligations(PROJECT_ID);
+            assertThat(ObligationsMap).isEmpty();
+        }
+
+        @Test
+        void obligationsInAProject() {
+            ProjectService mockInteractor = spy(interactor);
+            when(store.getProject(PROJECT_ID)).thenReturn(Optional.of(project));
+            final var dependencies = Set.of(dependency.setLicense("LICENSE"));
+            final var obligations = new HashMap<String, Set<Dependency>>();
+            obligations.put("Obligation", dependencies);
+            final var dependencyDto = DtoConverter.toBaseDto(dependency);
+
+            doReturn(obligationAnalyzer).when(mockInteractor).getObligationAnalyzerInstance(PROJECT_ID);
+            when(obligationAnalyzer.findObligations()).thenReturn(obligations);
+
+            final var obligationsResult = mockInteractor.getObligations(PROJECT_ID);
+            assertThat(obligationsResult.size()).isEqualTo(1);
```

Comment:
Not sure why you first check if there is one entry, and then access the entry.

Consider removing this line.